client/src/components/Layout/index.tsx

```diff
@@ -40,12 +71,29 @@ export function Header() {
         </Link>
 
         <NavigationBar />
-        <a
-          className="text-white hover:cursor-pointer hover:underline"
-          href="/api/logout"
-        >
-          Logout
-        </a>
+
+        {/* User dropdown */}
+        <div className="relative" ref={menuRef}>
+          <button
+            type="button"
+            onClick={() => setMenuOpen((prev) => !prev)}
+            className="font-public-sans hover:bg-blue-cool-70 flex items-center gap-2 rounded px-3 py-2 text-white focus:outline-none"
+          >
+            <Icon.Person size={3} aria-hidden />
+            {currentUser ? currentUser.username : 'Loading...'}
+          </button>
+
+          {menuOpen && (
+            <div className="ring-opacity-5 absolute right-0 mt-2 w-40 overflow-hidden rounded-md bg-white shadow-lg ring-1 ring-black">
+              <a
+                href="/api/logout"
+                className="font-public-sans block w-full px-4 py-2 text-sm text-gray-700 hover:bg-gray-100"
+              >
+                Logout
+              </a>
+            </div>
+          )}
```

Comment:
More a11y things - we should denote to screen readers that this is a menu opened by the button. We should also give the menu some structure using a list. 

I believe something like this works:
```suggestion
          {menuOpen && (
            <ul
              role="menu"
              className="ring-opacity-5 absolute right-0 mt-2 w-40 overflow-hidden rounded-md bg-white shadow-lg ring-1 ring-black"
            >
              <li role="none">
                <a
                  href="/api/logout"
                  role="menuitem"
                  className="font-public-sans block w-full px-4 py-2 text-sm text-gray-700 hover:bg-gray-100"
                >
                  Logout
                </a>
              </li>
            </ul>
          )}
```

Reply:
Disregard this if we go with my other suggestion to use the headless UI component.